app/classifier/index.cjsx

```diff
@@ -41,24 +39,30 @@ Classifier = React.createClass
 
   loadSubject: (subject) ->
     @setState subjectLoading: true
-    # TODO: Pre-cache subject images here and change loading state back.
+    preloadSubject subject
+      .then =>
+        if @props.subject is subject # The subject could have changed while we were loading.
+          @setState subjectLoading: false
+          @props.onLoad?()
 
   prepareToClassify: (classification) ->
     setTimeout => # TODO: Why is this here?
```

Comment:
Is this setTimeout still needed? I've been consistently getting a defined `classification` without it